opm/autodiff/WellInterface.hpp

```diff
@@ -195,8 +195,16 @@ namespace Opm
         virtual void addWellContributions(Mat&) const
         {}
 
-        virtual void addCellRates(RateVector& rates, int cellIdx) const
-        {}
+        void addCellRates(RateVector& rates, int cellIdx) const
```

Comment:
tiny nittpicking, please move this to the `_impl.hpp` file.

Reply:
I agree, we should follow the existing pattern here (although I see now that was broken recently in the ebos API merge, and I did not notice).